### ISSUE TITLE
Disabling Upload in TextField's value change when clearing the TextField value in SucceedListener fails

### ISSUE THREAD

**Issue by johannest** [vaadin/framework#11552]
- Vaadin Framework version 8.7.0
- Chrome
- Tomcat 8.5.3
- When disabling Upload button from TextField's value change listener (in case value is empty) in case the TextField value is cleared in the Upload.succeedListener, then upload remains enabled on the client-side, while in the server-side it is disabled.
- Example app
```
VerticalLayout vl = new VerticalLayout();
        vl.setSizeFull();

        Upload upload = new Upload("Textfield gets disabled after upload", new Upload.Receiver() {
            @Override
            public OutputStream receiveUpload(String filename, String mimeType) {
                System.out.println("Received upload");
                ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
                return byteArrayOutputStream;
            }
        });
        TextField textField = new TextField("Upload is enabled if this is not empty");
        textField.addValueChangeListener(e -> {
            upload.setEnabled(e.getValue() != null && !"".equals(e.getValue()));
        });
        upload.setEnabled(false);
        upload.addSucceededListener(e -> {
            textField.setValue("");
            // workaround:
            //JavaScript.eval("vaadin.forceSync();");
        });
        vl.addComponents(textField, upload);

        return vl;
```
Steps to reproduce the issue:

1. fill something to the textfield, and upload button is enabled
2. upload something, the textfield is cleared
3. after upload the upload button is enabled even when the textfield is empty
4. if you now upload something, server-side logs an error: `SEVERE: 
com.vaadin.server.UploadException: Warning: file upload ignored for 137 because the component was disabled`

Workaround: call `JavaScript.eval("vaadin.forceSync();");` after the TextField is cleared.

**Comment by TatuLund** [comment]
I think this needs to be retested with fix https://github.com/vaadin/framework/pull/11760

**Comment by TatuLund** [comment]
I retested this with 8.9.2 and issue did not occur anymore. It seems this was also fixed by #11760 as was anticipated.
